### DISCUSSION TITLE
How to *not interpolate* the input image for labeling

### DISCUSSION THREAD

**Discussion by notAstereo** [wkentaro/labelme#1516] [Ideas / Feature requests]
Hello,
I'd like to know if there is a way to tell labelme to *not interpolate* the image for labeling.
I want to see the individual pixels when I add my bounding boxes.

**Comment by gaspardringuenet** [comment]
Same issue here. Has someone found a way since?

**Reply by wkentaro** [reply]
currently not supported, but would be open to add as a feature given an motivating example. can you provide one? likely image + annotation target.

**Reply by gaspardringuenet** [reply]
My use case might be a bit peculiar as I am not working on natural images. I use LabelMe to annotate scientific data (here oceanographic sonar echogram i.e. spatio-temporal images showing the backscattering intensity). As illustrated in the second screenshot, strong intensity variations can occur at the pixel level, and they are biologically meaningful. Annotation can often be done despite the filter. However:

- In some case blurring prevents the user from seing the start / end of an object (here typically fish / other organisms assembled in a patch or layer)
- A gaussian filtering for this kind of data visualization is considered wrong as it alters real measurements

<figure>
   <img 
      width="504" 
      height="315" 
      alt="Screenshot 2026-06-09 at 15 12 38" 
      src="https://github.com/user-attachments/assets/4dc4b65f-c17b-48b0-a0cb-73af88f895a2" 
   />
   <figcaption>A example of an echogram image annotated using LabelMe. The x axis is ping time, the y axis is depth, and the intensity is mapped to the volume backscattering variable (a measure of the presence of organisms in the sampled volume)</figcaption>
</figure>

<figure>
   <img 
      width="504" 
      height="315" 
      alt="Screenshot 2026-06-09 at 15 12 49" 
      src="https://github.com/user-attachments/assets/1486f1c3-faea-4cc0-8fef-41dc06aa353e" />
   <figcaption>Zoom an a specific feature with high pixel-wise intensity variations</figcaption>
</figure>

**Reply by wkentaro** [reply]
interesting. can you please share the original image so that i can test myself (and address and verify the remedy)?

**Reply by gaspardringuenet** [reply]
Here it is !

<img width="6391" height="743" alt="image_0000" src="https://github.com/user-attachments/assets/047d1578-3920-46bc-9ddc-d54f5db4e17b" />

And here's a zoom on the region of interest I used as an example. It remains somehow blurry but this time it is linked to the underlying data.

<img width="1414" height="710" alt="image" src="https://github.com/user-attachments/assets/48947ced-d7dc-486c-842d-c8e9174d8e0c" />

**Reply by wkentaro** [reply]
looks i reproduced. will dig deeper soon 

<img width="1912" height="1242" alt="image" src="https://github.com/user-attachments/assets/f181c491-935c-437a-b3bd-85a4713ff1e3" />

**Reply by wkentaro** [reply]
> *This was generated by AI during triage.*

Tracked as #2142 (labeled `type: feature` + `ready-for-human`). The blur comes from the canvas always painting the image with the `SmoothPixmapTransform` render hint, so the image is bilinearly interpolated past 100% zoom. The fix is to add an opt-in nearest-neighbor mode for the image layer (shape outlines stay antialiased). @gaspardringuenet you can follow #2142 for progress.
